woocommerce/woocommerce-gutenberg-products-block
# Add product query support for Stock indicator block

Adds Product query support for the atomic stock indicator block(`woocommerce/product-stock-indicator`).

On the client side, when the Stock indicator block is used within the product query block, the markup will be rendered on the server side - No javascript related to Stock indicator block will be rendered.





Fixes #7333 




### Testing
1. Checkout this branch and run npm start
2. Be sure that you have latest version of the Gutenberg feature plugin installed and activated
3. Create a post/page and add the Product Query block.
4. Add Stock Indicator block to Post Template of product query block as shown in video below:

https://user-images.githubusercontent.com/16707866/203531286-f4dd88e8-299e-4bd1-b473-324b35c3c646.mov

5. Now save the post & confirm that:
    1. No JS related to the Product Stock Indicator block is loaded (search in your network tab for product-stock-indicator-frontend.js). As you can see in the screenshot below, product-stock-indicator-frontend.js showing before the changes made in this PR. This shouldn't be visible in network tab.
![Screenshot 2022-11-23 at 4 38 34 PM](https://user-images.githubusercontent.com/16707866/203532085-aea2f3d8-2175-423c-9601-7d8b5c1b94d5.png)
    2. UI of block should be identical in Frontend & Editor.
    3. That the settings are reflected correctly (eg: custom style). Try changing custom style from the sidebar.
    4. ADDITIONAL CSS CLASS(ES)(available in advanced toggle in sidebar) should be added to the container div
    5. Make changes to inventory & test the following status changes on the frontend side depending on the inventory of the product:
        - In stock
        - Out of stock
        - Available on backorder
        - 2 left in stock




### WooCommerce Visibility
